src/main/java/org/entando/entando/aps/system/services/pagemodel/PageModelService.java

```diff
@@ -294,6 +297,9 @@ protected BeanPropertyBindingResult validateDelete(PageModel pageModel) throws E
         if (references.size() > 0) {
             bindingResult.reject(PageModelValidator.ERRCODE_PAGEMODEL_REFERENCES, new Object[]{pageModel.getCode(), references}, "pageModel.cannot.delete.references");
         }
+        if (BooleanUtils.toBooleanDefaultIfNull(pageModel.isLocked(), false)) {
```

Comment:
this validation should be implemented inside `PageValidator`

Reply:
I agree with you. However, in this service class we have validation methods like validateAdd, validateEdit and validateDelete... To move the validateDelete to the `PageModelValidator` class I would need to move other methods and attributes used in this method. I want to avoid those big movements in these old classes. I am not sure if we would need to cherry pick, for example, changes from one branch to another. So, I added this new validation to the existing validateDelete method. What do you think?